openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -44,6 +44,15 @@
     with WAFFLE_SWITCHES.override(waffle.ESTIMATE_FIRST_ATTEMPTED, active=True):
         ...
 
+For long-lived flags, you may want to change the default for the flag from "off"
+to "on", so that it is "on" by default in devstack, sandboxes, or a new Open edX
+release. WaffleFlag has a DEPRECATED argument flag_undefined_default that we
+don't recommend you use any more. Instead, you could use a migration so that the
+flag appears in the Django admin.
+NOTE: There is a idea to have a “waffle.py” inside each app that lists these
```

Comment:
The primary reason I am hesitant about taking a stance on this is that we need to think through the repercussions of having this migration.  As a migration (or as a default), it effectively updates all environments, including open edX instances.  At that point, why not just remove the waffle flag entirely?  

If the only reason it is sticking around is to allow overrides by Open edX instances, then according to the Test Plan in the OEP, the waffle flag still needs to be tested in both on/off states on Master (but not on Stage).

Reply:
Maybe this wasn't clear, and I can update the comment, but the migration would only add the flag if it doesn't already exist.  So, it won't adjust existing flags, but will change the default for an environment for which the flag was never set.

We want devstacks and sandboxes to match what is in Production.  The case I have run into for several flags is that the flag is not yet ready to be removed (e.g., we still have some straggling course overrides, and in some cases, Product wants to provide one more Open edX release for people to move to the newer version), so we want to switch the default.

Additionally, for Open edX releases, we want to push toward the the default being on so that there will be less of an issue when the flag is removed.  If the owners of the Open edX instance is not yet ready for the change,  they can use the flag to do a longer roll-out before switching, still in advance of the flag being removed.

Reply:
I see.  Thanks. That explains the scenario more clearly for me.  For my cases, they were mostly all backend changes, which were Ok to be disabled on sandboxes and local devstacks, by default, during the interim time period (until the flag was removed).  Also, having environments with the flag disabled added confidence that the feature worked in that mode as well (while the flag still existed).

I do still think we need to think through (and test) the approach before suggesting it more widely.  What are the repercussions of having a migration?  Where would these migrations be kept?  What happens when the migration is rolled back?  If we instead go down the path of creating a management command for these, that will need further discovery and field-testing as well.

Reply:
I see two options of:
1. Leave this as-is and people will continue to use `flag_undefined_default` if they run into this issue, or
2. Deprecate it and add a note clarifying that there is investigation required to determine a better alternative (migrations, management command, or otherwise).

I don't want to deprecate it without a note, because so far, it is a valid solution that we have used.